src/statementsem.d

```diff
@@ -2975,13 +2975,14 @@ else
         else
         {
             /* Generate our own critical section, then rewrite as:
-             *  __gshared byte[CriticalSection.sizeof] critsec;
-             *  _d_criticalenter(critsec.ptr);
-             *  try { body } finally { _d_criticalexit(critsec.ptr); }
+             *  __gshared align(D_CRITICAL_SECTION.alignof) byte[D_CRITICAL_SECTION.sizeof] __critsec;
```

Comment:
I had `align((void*).sizeof)` before, but then switched to `align(D_CRITICAL_SECTION.alignof)` to clarify the intention instead of describing what happens. ;)